### PR TITLE
Add extra documentation for developers

### PR DESCRIPTION
**Pull request by samphilipd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,25 @@ cd elixir
 make clean test
 ```
 
+To recompile changed erlang files:
+
+```sh
+make compile
```
Comment (lexmag) [inline README.md]: `compile` compiles everything, not only Erlang files.

**Comment by lexmag** [comment]
Not sure that specifying these targets "scales" well. Maybe we should say "check Makefile for available targets" or create `help` target?

**Comment by josevalim** [comment]
Check makefile is definitely not an option. A help target can be a viable alternative though. In any case, those suggestions are way up in the README, they should be moved to the Contributing section IMO. :)

**Comment by samphilipd** [comment]
@lexmag I agree it's not scalable. But as a new contributor to Elixir it took me a while to figure out how to do these things, if they had been in the README it would have saved me a lot of time.

I am probably not the only person who has struggled with this and it's one small step to lowering the barrier of entry to making contributions :)

File: README.md [hunk 2]
```diff
@@ -77,6 +77,24 @@ bin/elixirc lib/elixir/lib/string.ex -o lib/elixir/ebin
 bin/elixir lib/elixir/test/elixir/string_test.exs
 ```
 
+To recompile only changed files (including erlang modules):
```
Comment (whatyouhide) [inline README.md]: Erlang should be capitalized.

File: README.md [hunk 3]
```diff
@@ -77,6 +77,24 @@ bin/elixirc lib/elixir/lib/string.ex -o lib/elixir/ebin
...
+
+```sh
+make compile
+```
+
+If you find yourself seeing errors during the bootstrap compilation step, you may need to reboostrap Elixir:
```
Comment (whatyouhide) [inline README.md]: Let's have a dash in re-bootstrap.

File: README.md [hunk 4]
```diff
@@ -77,6 +77,24 @@ bin/elixirc lib/elixir/lib/string.ex -o lib/elixir/ebin
 bin/elixir lib/elixir/test/elixir/string_test.exs
 ```
 
+To recompile only changed files (including Erlang modules):
```
Comment (josevalim) [inline README.md]: This always recompiles all and not only changed files.

Comment (josevalim) [inline README.md]: I think we should either say `make erlang` for compiling only Erlang or remove this altogether.

File: README.md [hunk 5]
```diff
@@ -77,6 +77,24 @@ bin/elixirc lib/elixir/lib/string.ex -o lib/elixir/ebin
...
+
+```sh
+make compile
+```
+
+If you find yourself seeing errors during the bootstrap compilation step, you may need to re-boostrap Elixir:
```
Comment (josevalim) [inline README.md]: "If your contribution fails the build during the bootstrapping of the language, you can reproduce it locally by deleting all of Elixir beam files and compiling again:"

File: README.md (hunk 4 above)
Comment (samphilipd) [inline README.md]: Are you sure? When I run `make compile` on my machine it only outputs recompilation for changed files.

`➜  elixir git:(master) make clean`

```
➜  elixir git:(master) make compile
==> elixir (compile) # takes a couple of minutes
Compiled src/elixir_parser.yrl
Compiled src/elixir.erl
... lots more ...
```

And again:

```
➜  elixir git:(master) make compile
==> elixir (compile) # takes a couple of seconds
➜  elixir git:(master)
```

Comment (josevalim) [inline README.md]: @samphilipd if you change a single file in the elixir application, the whole application will be recompiled. The same for mix, eex, etc.

**Comment by eksperimental** [comment]
I am for having a `help` target that documents everything.

If we are going to go "public", I think we should renamed target "clean_exbeam" as it is doesn't follow the rule of separating words with underscores. How about "clean_elixir_beam(s)"?

**Comment by ericmj** [comment]
Renaming `clean_exbeam` is probably fine but we should careful when renaming Makefile targets since a lot of build scripts rely on those names.

**Comment by josevalim** [comment]
make help will be a welcome addition. Let's consider it separate from this
PR.
-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by josevalim** [comment]
@samphilipd I think you messed something up. :D

**Comment by samphilipd** [comment]
@josevalim rebase is a dangerous tool... it should be fixed now.

File: README.md [hunk 6]
```diff
@@ -77,6 +77,28 @@ bin/elixirc lib/elixir/lib/string.ex -o lib/elixir/ebin
...
+make clean_exbeam compile
+```
+
+Or to rebuild everything from scratch without running tests:
+
+```sh
+make clean compile
+```
+
+More tasks can be found by reading the [Makefile](./Makefile)
```
Comment (josevalim) [inline README.md]: Missing `.` at the end of the sentence (sorry!).

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**